src/LinAlg/matrix.h

```diff
@@ -129,7 +129,8 @@ namespace utl //! General utility classes and functions.
     vector<T>& operator-=(const vector<T>& X) { return this->add(X,T(-1)); }
     //! \brief Add the given vector \b X scaled by \a alfa to \a *this.
     vector<T>& add(const std::vector<T>& X, const T& alfa = T(1),
-                   int ofsx = 0, int stridex = 1);
+                   unsigned int ofsx = 0, unsigned int stridex = 1,
+                   unsigned int ofsy = 0, unsigned int stridey = 1);
```

Comment:
Actually, negative stride should be allowed (in combination of non-negative offset), in case someone wants to loop backward in an array. Think some BLAS routines accounts for that at least.

Reply:
right. changed back to signed.